sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -846,6 +853,8 @@ abstract class ToTimestamp
               case NonFatal(_) => null
             }
           }
+        case LongType =>
+          Math.multiplyExact(t.asInstanceOf[Long], scaleFactor.asInstanceOf[Long])
```

Comment:
@cloud-fan Do we do range checks on timestamps? (Do we even have published valid ranges?) Because if we do, then this could produce valid Long values that are out of range.